### DISCUSSION TITLE
Custom Edge Agent

### DISCUSSION THREAD

**Discussion by AbhinavSharmaSD** [aws/aws-iot-device-sdk-js-v2#530] [Q&A]
Hi,
The **aws-iot-device-sdk-js-v2** is provided as a sample edge agent for testing purposes. However, if we want to create our own custom Edge Agent, what steps should we follow, and how can we customize it to fit our specific requirements?

**Comment by jmklix** [comment]
I would suggest looking at this [Edge Agent Developer Guide](https://github.com/aws/aws-iot-fleetwise-edge/blob/main/docs/dev-guide/edge-agent-dev-guide.md) if you haven't already

>...
> The Reference Implementation for AWS IoT FleetWise ("FWE") provides C++ libraries that can be run with simulated vehicle data on certain supported vehicle hardware or that can help you develop an Edge Agent to run an application on your vehicle that integrates with AWS IoT FleetWise. You can use AWS IoT FleetWise pre-configured analytic capabilities to process collected data, gain insights about vehicle health, and use the service's visual interface to help diagnose and troubleshoot potential issues with the vehicle.
>...
